src/main/java/spypunk/tetris/ui/view/TetrisSettingsView.java

```diff
@@ -0,0 +1,557 @@
+                        && !(evt.getKeyChar() == KeyEvent.VK_DELETE || evt.getKeyChar() == KeyEvent.VK_BACK_SPACE)) {
+                    getToolkit().beep();
+                    evt.consume();
+                }
+            }
+
+            public void keyReleased(KeyEvent e) {
+                textSetter(e, tpause);
+            }
+        });
+
+        mute = new JLabel("MUTE");
+        mute.setFont(fontCache.getDefaultFont());
+        mute.setForeground(Color.WHITE);
+        tmute = new JTextField(KeyEvent.getKeyText(controls.get(6)));
+        tmute.setFont(fontCache.getDefaultFont());
+        tmute.setForeground(Color.RED);
+        tmute.setBackground(Color.black);
+        tmute.setCaretColor(Color.WHITE);
+        tmute.addKeyListener(new java.awt.event.KeyAdapter() {
+            public void keyTyped(java.awt.event.KeyEvent evt) {
+                if (tmute.getText().length() >= 1
+                        && !(evt.getKeyChar() == KeyEvent.VK_DELETE || evt.getKeyChar() == KeyEvent.VK_BACK_SPACE)) {
+                    getToolkit().beep();
+                    evt.consume();
+                }
+            }
+
+            public void keyReleased(KeyEvent e) {
+                textSetter(e, tmute);
+            }
+        });
+
+        incvolume = new JLabel("INC. VOL");
+        incvolume.setFont(fontCache.getDefaultFont());
+        incvolume.setForeground(Color.WHITE);
+        tincvolume = new JTextField(KeyEvent.getKeyText(controls.get(7)));
+        tincvolume.setFont(fontCache.getDefaultFont());
+        tincvolume.setForeground(Color.RED);
+        tincvolume.setBackground(Color.black);
+        tincvolume.setCaretColor(Color.WHITE);
+        tincvolume.addKeyListener(new java.awt.event.KeyAdapter() {
+            public void keyTyped(java.awt.event.KeyEvent evt) {
+                if (tincvolume.getText().length() >= 1
+                        && !(evt.getKeyChar() == KeyEvent.VK_DELETE || evt.getKeyChar() == KeyEvent.VK_BACK_SPACE)) {
+                    getToolkit().beep();
+                    evt.consume();
+                }
+            }
+
+            public void keyReleased(KeyEvent e) {
+                textSetter(e, tincvolume);
+            }
+        });
+
+        decvolume = new JLabel("DEC. VOL");
+        decvolume.setFont(fontCache.getDefaultFont());
+        decvolume.setForeground(Color.WHITE);
+        tdecvolume = new JTextField(KeyEvent.getKeyText(controls.get(8)));
+        tdecvolume.setFont(fontCache.getDefaultFont());
+        tdecvolume.setForeground(Color.RED);
+        tdecvolume.setBackground(Color.black);
+        tdecvolume.setCaretColor(Color.WHITE);
+        tdecvolume.addKeyListener(new java.awt.event.KeyAdapter() {
+            public void keyTyped(java.awt.event.KeyEvent evt) {
+                if (tdecvolume.getText().length() >= 1
+                        && !(evt.getKeyChar() == KeyEvent.VK_DELETE || evt.getKeyChar() == KeyEvent.VK_BACK_SPACE)) {
```

Comment:
Neden tüm listenerlarda VK_DELETE ve VK_BACK_SPACE e karşı bir kontrol yapıyoruz?

Reply:
Hocam default kontrollerde örneğin sola gitme tuşu sol ok. Bunu ekrana "left" şeklinde yazdırdım. Bunu değiştirmek için klavyeden herhangi bir harf girileceğinde sadece silinebilmesi ve tek bir harf girilebilmesi için bu kontrolü yaptım.